client-haskell/src/Icepeak/Client.hs

```diff
@@ -48,9 +53,14 @@ import qualified Network.HTTP.Client as HTTP
 import qualified Network.HTTP.Types.Status as HTTP
 import qualified Network.HTTP.Types.Header as Header
 import qualified Network.HTTP.Types.URI as URI
+import qualified Control.Retry as Retry
 
 -- | Config paired with HTTP manager for making requests.
-data Client = Client HTTP.Manager Config
+data Client m = Client 
+  { clientHttpManager :: HTTP.Manager
+  , clientConfig      :: Config
+  , clientRetryPolicy :: Maybe (RetryPolicyM m)
```

Comment:
Nothing and Just doNotRetry seem to encode the same information. I suggest making this field not Maybe, and exporting doNotRetry. That would simplify both the implementation and the usage of the library.

Reply:
Yes, this seems like a good idea!